STL_Extension/include/CGAL/Container_helper.h

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2014  GeometryFactory (France).  All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org)
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: LGPL-3.0-or-later OR LicenseRef-Commercial
+// 
+//
+// Author(s)     : Andreas Fabri
+
+#ifndef CGAL_CONTAINER_HELPER_H
+#define CGAL_CONTAINER_HELPER_H
+
+#include <CGAL/array.h>
+#include <CGAL/Has_member.h>
+
+#include <CGAL/assertions.h>
+#include <boost/mpl/logical.hpp>
+#include <boost/utility/enable_if.hpp>
+
+#include <cstddef>
+
+namespace CGAL {
+namespace internal {
+
+CGAL_GENERATE_MEMBER_DETECTOR(size);
+CGAL_GENERATE_MEMBER_DETECTOR(resize);
+
+// Typical container
+template <class Container>
+void resize(Container& c, std::size_t size,
+            typename boost::enable_if_c<has_resize<Container>::value>::type* = nullptr)
+{
+  c.resize(size);
+}
+
+// Container without a resize() function, but with a size() function (e.g. an array)
+template <class Container>
+void resize(Container& CGAL_assertion_code(array), std::size_t CGAL_assertion_code(size),
+            typename boost::enable_if<
+              boost::mpl::and_<
+                boost::mpl::not_<has_resize<Container> >,
+                                 has_size<Container> > >::type* = nullptr)
+{
+  CGAL_assertion(array.size() == size);
+}
+
+// A class with neither resize() nor size(), can't enforce size (it better be correct!)
+template <class Container>
+void resize(Container&, std::size_t,
+            typename boost::disable_if<
+              boost::mpl::or_<has_resize<Container>,
+                              has_size<Container> > >::type* = nullptr)
+{
+}
+
+// Plenty of times we ask for a model of RandomAccessContainer and use reserve, but some models
+// of the concept do not have a reserve (e.g. deque)
+CGAL_GENERATE_MEMBER_DETECTOR(reserve);
+
+// Container with 'reserve'
+template <class Container>
+void reserve(Container& c, std::size_t size,
```

Comment:
Wouldn't mind some feedback on that one. 

I added reserve here (size/resize was also added by me a while ago in IO for similar reasons). because I noticed that the concepts are requiring say RandomAccessContainer, but the models do not necessarily have a reserve function (a lot would fail if we provided strict concepts and not std::vector). However, when it's there, it's nice to be able to use it. Therefore this addition, but it's a bit awkward.

Reply:
I think it's fine. It's just hard to extend if the user comes with its own container (because it would mean defining an overload) but `reserve()` is a pretty good name so for most users it will be fine. An alternative could be to provide those function from a documented struct but I don't think it does worth it.You might want to rename internal into container_helper just to make easier to match to this header when reading the implementation.